### PR TITLE
fix: toolbox panel wasn't updated on switching tools via shortcuts

### PR DESCRIPTION
**Pull request by rodolforg**
Related to: PR #3012 (313878d4d12a32f24b533d0832a6660b63ee149a)

### CODE REVIEW

File: synfig-studio/src/gui/docks/dock_toolbox.cpp [hunk 1]
```diff
@@ -146,10 +146,23 @@ void Dock_Toolbox::read_layout_string(const std::string& params) const
...
+		for (const auto& item : state_button_map) {
+			if (item.first == statename && !item.second->get_active()) {
+				item.second->set_active(true);
+				break;
+			}
+		}
+	} catch (...) {
+		changing_state_ = false;
+		throw;
+	}
```
Comment (Keyikedalube) [inline synfig-studio/src/gui/docks/dock_toolbox.cpp:163]: This `catch (...)` block is for string "none", right? So the program don't crash...

**Review by Keyikedalube — APPROVED** [review]

File: synfig-studio/src/gui/docks/dock_toolbox.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-studio/src/gui/docks/dock_toolbox.cpp:163]: Well, I just copied from original code before it was deleted XD https://github.com/synfig/synfig/commit/313878d4d12a32f24b533d0832a6660b63ee149a

Comment (Keyikedalube) [reply synfig-studio/src/gui/docks/dock_toolbox.cpp:163]: The code is very safe then 😂

Because I didn't exactly get what try block is catching LOL.

Comment (rodolforg) [reply synfig-studio/src/gui/docks/dock_toolbox.cpp:163]: I think it's because `set_state()`, that deals with GUI stuff... Oh well, I'll leave it as is XD

**Pull request merged**